testsuite/tests/libratest/smoke_test.rs

```diff
@@ -1312,3 +1327,185 @@ fn test_key_manager_consensus_rotation() {
         );
     }
 }
+
+struct AccountContext {
+    public_key: Ed25519PublicKey,
```

Comment:
True, because it was for testing, I went through and did it this way, but it's perfectly fine to not copy around a second key and derive it from the private.